kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -365,6 +371,43 @@
       "Whether to enable REST Proxy V3 API. Default is true.";
   private static final boolean API_V3_ENABLE_DEFAULT = true;
 
+  public static final String RATE_LIMIT_ENABLE_CONFIG = "rate.limit.enable";
+  private static final String RATE_LIMIT_ENABLE_DOC =
+      "Whether to enable request rate-limiting. Default is false.";
+  private static final boolean RATE_LIMIT_ENABLE_DEFAULT = false;
+
+  public static final String RATE_LIMIT_BACKEND_CONFIG = "rate.limit.backend";
+  private static final String RATE_LIMIT_BACKEND_DOC =
+      "The rate-limiting backend to use. The options are 'guava' and 'resilience4j'. Default is "
+          + "'guava'.";
+  private static final String RATE_LIMIT_BACKEND_DEFAULT = "guava";
+
+  public static final String RATE_LIMIT_PERMITS_PER_SEC_CONFIG = "rate.limit.permits.per.sec";
+  private static final String RATE_LIMIT_PERMITS_PER_SEC_DOC =
+      "The maximum number of permits to emit per second. A permit is an unit of cost for a "
+          + "request. More expensive requests will consume more permits. The cost for each "
+          + "resource/method can configured via rate.limit.default.cost and rate.limit.costs. "
+          + "Default is 50.";
+  private static final Integer RATE_LIMIT_PERMITS_PER_SEC_DEFAULT = 50;
+
+  public static final String RATE_LIMIT_TIMEOUT_MS_CONFIG = "rate.limit.timeout.ms";
+  private static final String RATE_LIMIT_TIMEOUT_MS_DOC =
+      "How much to wait for the necessary permits to proceed, in milliseconds. A request that "
```

Comment:
This is like the grace period for Produce. I was wondering if it should have "grace" in the variable name to indicate that (although I'm 50:50 on that).

The text should say "How much time to wait for the necessary...."  or "How long to wait for the necessary.... "

Reply:
This is not like the grace period. The grace period allows requests to proceed even though they are over the limit, for a small amount of time. This timeout actually blocks the request from proceeding until the rate is below the limit.